nvidia/lib

```diff
@@ -9,12 +9,16 @@ ARCH_TRIPLET="${U_MACHINE}-${U_KERNEL,,}-${U_USERLAND,,}"
 
 NVIDIA_SUPPORT_DISABLED="$(snapctl get nvidia-support.disabled)"
 
-if [ -L "/var/lib/snapd/hostfs/usr/lib/${ARCH_TRIPLET}/libcuda.so" ] ; then
-    NVIDIA_SUPPORT_CLASSIC="true"
+NVIDIA_SUPPORT_CLASSIC="false"
+[ -L "/var/lib/snapd/hostfs/usr/lib/${ARCH_TRIPLET}/libcuda.so" ] && NVIDIA_SUPPORT_CLASSIC="true"
+
+if snapctl is-connected gpu-2404 || snapctl is-connected graphics-core22; then
+    NVIDIA_SUPPORT_CORE="true"
 else
-    NVIDIA_SUPPORT_CLASSIC="false"
+    NVIDIA_SUPPORT_CORE="false"
 fi
```

Comment:
With this, both NVIDIA_SUPPORT_CLASSIC and NVIDIA_SUPPORT_CORE can be true, when on classic with a provider installed and (auto-)connected. This is correctly handled in `nvidia/gpu-2404-optional-wrapper` because of the `exec` exit points but not in `nvidia/nvidia-container-toolkit`:


```console
$ snap connections docker
Interface          Plug                     Slot                                 Notes
...
content[gpu-2404]  docker:gpu-2404          mesa-2404:gpu-2404                   -
```


```console
$ snap logs docker.nvidia-container-toolkit -n 100
2025-06-17T18:19:32+02:00 docker.nvidia-container-toolkit[109211]: Running on Classic system
2025-06-17T18:19:32+02:00 docker.nvidia-container-toolkit[109211]: Running on Ubuntu Core system
```